### PR TITLE
Add support for BigQuery

### PR DESCRIPTION
**Pull request by epsstan**
| Q                       | A                                |
| ----------------------- | -------------------------------- |
| Fixed issues?           |  |
| Tests added + passed?   | Yes                              |
| Changeset added?        | Yes                              |
| Documentation PR link   |                                  |
| Any dependency changes? | No                               |

### CODE REVIEW

File: .changeset/bigquery.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+'@finos/legend-studio': patch
```
Comment (akphi) [inline .changeset/bigquery.md:1]: Could you use the script to generate this. This file is not of front matter format and will not be parsed properly

File: packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx [hunk 2]
```diff
@@ -604,6 +636,19 @@ const SnowflakePublicAuthenticationStrategyEditor = observer(
   },
 );
 
+const GCPApplicationDefaultCredentialsAuthenticationStrategyEditor = observer(
```
Comment (akphi) [inline packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx:639]: If you just return nothing like this, perhaps, you don't need this at all, the top handler will render nothing when the auth strategy is not supported

File: packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx [hunk 3]
```diff
@@ -1080,6 +1132,13 @@ const renderAuthenticationStrategyEditor = (
         isReadOnly={isReadOnly}
       />
     );
+  } else if (authSpec instanceof GCPApplicationDefaultCredentialsAuthenticationStrategy) {
```
Comment (akphi) [inline packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx:1135]: refer to comment above

File: packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/helpers/V1_RelationalConnectionBuilderHelper.ts [hunk 4]
```diff
@@ -123,6 +127,16 @@ export const V1_processDatasourceSpecification = (
     snowflakeSpec.quotedIdentifiersIgnoreCase =
       protocol.quotedIdentifiersIgnoreCase;
     return snowflakeSpec;
+  } else if (protocol instanceof V1_BigQueryDatasourceSpecification) {
+        assertNonEmptyString(
+          protocol.projectId,
+          'BigQuery datasource specification property project id is missing',
```
Comment (akphi) [inline packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/helpers/V1_RelationalConnectionBuilderHelper.ts:133]:
```suggestion
          'BigQuery datasource specification property project ID is missing',
```

File: packages/legend-studio/src/models/protocols/pure/v1/transformation/pureProtocol/serializationHelpers/V1_ConnectionSerializationHelper.ts [hunk 5]
```diff
@@ -167,6 +170,15 @@ const snowflakeDatasourceSpecificationModelSchema = createModelSchema(
   },
 );
 
+const bigqueryDatasourceSpecificationModelSchema = createModelSchema(
+  V1_BigQueryDatasourceSpecification,
+  {
+    _type: usingConstantValueSchema(V1_DatasourceSpecificationType.BIGQUERY),
+    projectId: primitive(),
```
Comment (akphi) [inline packages/legend-studio/src/models/protocols/pure/v1/transformation/pureProtocol/serializationHelpers/V1_ConnectionSerializationHelper.ts:177]: Please sort this alphabetically

File: packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/helpers/V1_RelationalConnectionBuilderHelper.ts [hunk 6]
```diff
@@ -123,6 +127,16 @@ export const V1_processDatasourceSpecification = (
     snowflakeSpec.quotedIdentifiersIgnoreCase =
       protocol.quotedIdentifiersIgnoreCase;
     return snowflakeSpec;
+  } else if (protocol instanceof V1_BigQueryDatasourceSpecification) {
+        assertNonEmptyString(
```
Comment (akphi) [inline packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/helpers/V1_RelationalConnectionBuilderHelper.ts:131]: Where is the assertion for `defaultDataset`?

Comment (epsstan) [reply packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/helpers/V1_RelationalConnectionBuilderHelper.ts]: It is an optional property. If specified it is used as the default dataset.

File: .changeset/bigquery.md (hunk 1 above)
Comment (epsstan) [reply .changeset/bigquery.md]: Done

File: packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/helpers/V1_RelationalConnectionBuilderHelper.ts (hunk 4 above)
Comment (epsstan) [reply packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/to/helpers/V1_RelationalConnectionBuilderHelper.ts]: Done

File: .changeset/yellow-students-tie.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+"@finos/legend-studio-plugin-query-builder": patch
+"@finos/legend-studio-plugin-tracer-zipkin": patch
+"@finos/legend-studio-preset-dsl-text": patch
+"@finos/legend-studio-preset-dummy": patch
+"@finos/legend-studio-preset-external-format-json-schema": patch
+"@finos/legend-studio-shared": patch
+"@finos/stylelint-config-legend-studio": patch
+---
+
+Add support for BigQuery
```
Comment (akphi) [inline .changeset/yellow-students-tie.md:20]: Given how urgent you need this, we will merge this, but please prune the changeset next time.

Comment (akphi) [reply .changeset/yellow-students-tie.md:20]: We would have 2 changets: 1 for your change `Add support for BigQuery` for only `@finos/legend-studio`, 1 for all the changes across modules

**Review by akphi — APPROVED** [review]

**Comment by akphi** [comment]
We should come back and add some small roundtrip test for this

**Pull request merged**